internal/pkg/instrumentation/probe/probe.go

```diff
@@ -36,17 +33,36 @@ type Probe interface {
 	// the information about the package the Probe instruments.
 	Manifest() Manifest
 
-	// Load loads all the eBPF programs and maps required by the Probe.
-	// It also attaches the eBPF programs to the target process.
-	// TODO: currently passing Sampler as an initial configuration - this will be
-	// updated to a more generic configuration in the future.
-	Load(*link.Executable, *process.Info, *sampling.Config) error
+	// Init initializes the Probe, setting up the reader, closers, and sampling config.
+	Init(*sampling.Config) error
 
 	// Run runs the events processing loop.
 	Run(func(ptrace.ScopeSpans))
 
 	// Close stops the Probe.
 	Close() error
+
+	// GetLogger returns the *slog.Logger associated with the Probe.
+	GetLogger() *slog.Logger
+
+	// Spec returns the *ebpf.CollectionSpec for the Probe.
+	Spec() (*ebpf.CollectionSpec, error)
+
+	// SetCollection sets the *ebpf.Collection for the Probe.
+	SetCollection(*ebpf.Collection)
+
+	// GetCollection returns the *ebpf.Collection for the Probe.
+	GetCollection() *ebpf.Collection
+
+	// GetUprobes returns a list of *Uprobes for the Probe.
+	GetUprobes() []*Uprobe
+
+	// GetConsts returns a list of Consts for the Probe.
+	GetConsts() []Const
```

Comment:
These are duplicates of the `Manifest` method which contains these values.

Reply:
True, and it wasn't clear to me because we change the name of them in the Manifest: `Uprobes`->`FunctionSymbol` and `Consts`->`StructFieldIDs`

The latter I think is because the Manifest doesn't actually store all the Consts in the Probe, just the consts for fields. If we're loading the uprobes in the manager now and injecting the consts there, we'll need all of the consts.

Do you think we should refactor the Manifest struct a bit to align better with the Probe definition?

Reply:
We should definitely unify these types respectively.

Given they are static to a probe, keeping their definitions in the `Manifest` seems appropriate.

Reply:
@MrAlias looking more at the Manifest, I think this change in particular makes it redundant and everything it provides can be in the Probe itself. Given that we are defining the Probe as a more static representation now, I pushed an update to remove the Manifest type entirely and just use the references from it directly in the Probe.

Let me know if there's some other reason for Manifest I'm missing